packages/path-provider/lib/path_provider.dart

```diff
@@ -40,3 +40,16 @@ const _channel = const MethodChannel('plugins.flutter.io/path_provider');
       return null;
     return new Directory(path);
   }
+
+  /// Path to a directory where the application may access top level storage
+  ///
+  /// On iOS, This function returns null as it is not possible to access outside
+  /// the apps sandbox
+  ///
+  /// On Android this returns getExternalStorageDirectory
+  Future<Directory> getExternalStorageDirectory() async {
```

Comment:
On iOS, this will result in a [MissingPluginException](https://docs.flutter.io/flutter/services/MissingPluginException-class.html) since there is no handling of  a method with that name. To avoid this, you could [check the OS](https://docs.flutter.io/flutter/dart-io/Platform-class.html) here and throw for instance a [StateError](https://www.dartdocs.org/documentation/flutter/0.0.21/dart-core/StateError-class.html). Remember to document also, that the OS needs to be determined before calling this method.